chapters/publication.tex

```diff
@@ -1,96 +1,171 @@
+Thinking of the writing process in this way
+is intended to improve organization and reduce error,
+such that there is no risk of materials being compiled
+with out-of-date results, or of completed work being lost or redundant.
+
+\subsection{Dynamic documents}
+
+Dynamic documents are a broad class of tools that enable such a workflow.
+The term ``dynamic'' can refer to any document-creation technology
+that allows the creation of explicit references to raw output files.
+This means that, whenever outputs are updated,
+the next iteration of the document will automatically include
+all changes made to all outputs without any additional intervention from the writer.
+This means that updates will never be accidentally excluded,
+and it further means that updating results will not become more difficult
+as the number of inputs grows,
+because they are all managed by a single integrated process.
+
+You will note that this is not possible in tools like Microsoft Office.
+In Word, for example, you have to copy and paste each object individually
+whenever tables, graphs or other inputs have to be updated.
+This means that both the features above are not available:
+fully updating the document becomes more and more time-consuming
+as the number of inputs increases,
+and it therefore becomes more and more likely
+that a mistake will be made or something will be missed.
+Furthermore, it is very hard to simultaneously edit or track changes
+in a Microsoft Word document.
+It is usually the case that a file needs to be passed back and forth
+and the order of contributions strictly controlled
+so that time-consuming resolutions of differences can be avoided.
+Therefore this is a broadly unsuitable way to prepare technical documents.
+
+There are a number of tools that can be used for dynamic documents.
+They fall into two broad groups --
+the first compiles a document as part of code execution,
+and the second operates a separate document compiler.
+In the first group are tools such as R's RMarkdown and Stata's \texttt{dyndoc}.
+These tools ``knit'' or ``weave'' text and code together,
+and are programmed to insert code outputs in pre-specified locations.
+Documents called ``notebooks'' (such as Jupyter) work similarly,
+as they also use the underlying analytical software to create the document.
+These types of dynamic documents are usually appropriate for short or informal materials
+because they tend to offer limited editability outside the base software
+and often have limited abilities to incorporate precision formatting.
+
+On the other hand, some dynamic document tools do not require
+operation of any underlying software, but simply require
+that the writer have access to the updated outputs.
+One very simple one is Dropbox Paper, a free online writing tool
+that allows linkages to files in Dropbox,
+which are then automatically updated anytime the file is replaced.
+Like the first class of tools, Dropbox Paper has very limited formatting options,
```

Comment:
What is the "_first class of tools_"? We haven't mention classes of tools in the text above where this seems to be a reference to.

Reply:
no clue. 2820ec58fc7479c200b72c4e87f6fbaa4ebb11b1